.github/workflows/main.yml

```diff
@@ -0,0 +1,31 @@
+name: Publish to FTP on Development Push
+
+on:
+  push:
+    branches:
+      - development
+
+jobs:
+  publish:
+    runs-on: windows-latest
```

Comment:
i created two environments: test and production. You need to specify what environment this is for, so the workflow knows what variables/secrets to use.
https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment